### ISSUE TITLE
Hashes nested inside of arrays not properly parsed in query params / body.

### ISSUE THREAD

**Issue by sealabcore** [jnunemaker/httparty#574]
Example hash payload that does not parse properly:

```ruby
{:marketing_event => { :marketed_resources=>[{:type=>"product", :id=>57474842640}]}}
```

The resulting string is `marketing_event[marketed_resources][][type]=product`. You can see that the `id` is completely omitted from the resulting string.

This could have been a result of the refactor of this class in the latest version.
https://github.com/jnunemaker/httparty/blob/master/lib/httparty/hash_conversions.rb

**Comment by jnunemaker** [comment]
Thanks for the report. I'd be up for a PR. I can try to tackle it at some point, but I'm not sure when. If anyone else has time, let me know.

**Comment by jmortlock** [comment]
Hi @jnunemaker I have add a PR which fixes this, let me know if you need anything extra to get this merged.

**Comment by jnunemaker** [comment]
Closed by #583
